readalongs/text/convert_xml.py

```diff
@@ -100,53 +97,37 @@ def convert_word(word: str, lang: str):
                 well as in the final output.
         """
 
-        if lang == "eng":
-            # Hack to use old English LexiconG2P
-            # Note: adding eng_ prefix to vars that are used in both blocks to make mypy
-            # happy. Since the two sides of the if and in the same scope, it complains about
-            # type checking otherwise.
-            if "eng-arpabet" not in output_orthography:
-                raise ValueError(
-                    f'Cannot g2p "eng" to output orthography "{output_orthography}".'
-                )
-            eng_converter = getLexiconG2P(
-                os.path.join(os.path.dirname(LEXICON_PATH), "cmu_sphinx.metadata.json")
-            )
-            try:
-                eng_text, _ = eng_converter.convert(word)
-                eng_valid = is_arpabet(eng_text)
-            except KeyError as e:
+        try:
+            converter = make_g2p(lang, output_orthography, tokenize=False)
+        except InvalidLanguageCode as e:
+            raise ValueError(
+                f'Could not g2p "{word}" from "{lang}" to "{output_orthography}": {e} '
+                f'\nRun "readalongs langs" to list languages supported by ReadAlongs Studio.'
+            ) from e
+        except NoPath as e:
+            raise ValueError(
+                f'Could not g2p "{word}": no path from "{lang}" to "{output_orthography}".'
+                f'\nRun "readalongs langs" to list languages supported by ReadAlongs Studio.'
+            ) from e
+        tg = converter(word)
+        text = tg.output_string
+        if not text:
+            # TODO Convert this to a check of the kind "is the mapping a lexicon-based one".
```

Comment:
Hmm.  This should be a method on the `Transducer`  base class in g2p, but you can do it like this:

```python
if "lexicon" in (t.mapping.kwargs.get("type") for t in converter.transducers):
```

Reply:
Easy enough, thanks for the hint!

Reply:
Done. I love how your refactoring in your dev.alignments PR, making Transducer and CompositeTransducer compatible, makes this code so much simpler, with `coverter.transducers` just doing the right thing in each case.

Reply:
Hey, I just had a thought: an empty g2p output from any language, lexicon or rule based, is really an error situation that will cause soundswallower to fail. I could simply that code a fair bit to just
```
        if not text:
            nonlocal g2p_empty_warning_count
            if g2p_empty_warning_count < 2 or verbose_warnings:
                g2p_empty_warning_count += 1
                LOGGER.warning(
                    f'The output of the g2p process for "{word}" with lang "{lang}" is empty.'
                )
        valid = converter.check(tg, shallow=True) and text
```

Reply:
The useful side effect here is that if a language's rules turn a token to nothing, it'll be tried again with und and thus we'll be much more likely to get a successful alignment.